### ISSUE TITLE
Normalizing features df

### ISSUE THREAD

**Issue by martimpassos** [damoncrockett/ivpy#95]
Hi Damon, this is more of a theoretical question than an issue, sorry if this isn't the best channel to do so: I have a pickled feature DF of about 250k images. Whenever I get a similarity search request, I `extract('neural')` only the query image, append it to the pickled df and perform the search using `nearest`. Is it necessary to `norm` the df when appending a single image? It takes about 3min to do so every time, and I suspect the difference is negligible... I don't pickle the features df again after the process (thus discarding the query image)

**Comment by damoncrockett** [comment]
@martimpassos That's a good question. I think there are two ways you could do this without renorming every time: (1) just don't normalize in the first place, and make sure you use the same neural extraction model each time. I should note that I just changed the model from a tensorflow version to a PyTorch version of ResNet50 (Tensorflow sucks and I've had way too much trouble with version compatibility issues). Theoretically, it should be the same model, but I noticed that the tf version has 2048 dimensions, and the PyTorch only 1000. I think this is because with tf, I used the penultimate layer, and with PyTorch, I'm using the last layer, but without softmax (softmax vectors are not usable as position vectors, I've learned). So, if you want to keep using the old tf model, you should open your own git branch or just never pull any new changes to ivpy.

(2) If you need to store a normed feature df, you could then also just store the mins and maxes for each column, and when a new vector comes in, just norm it using those extrema. That way, you don't have to renorm the whole df, and it should be near instantaneous.
